### ISSUE TITLE
Using AWS.jl to connect NOT to AmazonS3

### ISSUE THREAD

**Issue by akubisch** [JuliaCloud/AWS.jl#111]
We have set up a local endpoint on a server here using S3 storage. I am currently trying to connect to it, setting the `ep` to the server url  (like `https://s3-test.myhost.net`) and setting the correct access and secret key. Yet, I just get the result that the access key is unknown and apparently the responding server was again AmazonS3. Is there any way of not directing the request to amazon? The following is what I tried:
```
using AWS
using AWS.S3

env = AWSEnv(id="XXXXXXXXXXXXX",
            key="XXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            ep="https://s3-test.myhost.net")

S3.list_all_buckets(env)

```
What finally results in this:
```
S3Response
content_length: 0
date: Wed, 21 Feb 2018 14:38:20 GMT
server: AmazonS3
eTag:
http_code: 403
delete_marker: false
id_2:
request_id:
version_id:
headers: Pair{String,String} Dict
    Transfer-Encoding => chunked
    status_code => 403
    http_major => 1
    x-amz-request-id => XXXXXXXXXXXXX
    Content-Type => application/xml
    Date => Wed, 21 Feb 2018 14:38:20 GMT
    http_minor => 1
    Keep-Alive => 1
    Server => AmazonS3
    x-amz-id-2 => XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX=
obj: AWS.S3.S3Error("InvalidAccessKeyId", "The AWS Access Key Id you provided does not exist in our records.", "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX=", "XXXXXXXXXXXXX")
pd: Not set
```

Thank you a lot in advance for your help!

**Comment by samoconnor** [comment]
The AWSS3.jl package has an `:endpoint` option that allows the endpoint to be overridden: https://github.com/samoconnor/AWSS3.jl/blob/master/src/AWSS3.jl#L91

**Comment by tanmaykm** [comment]
It's a bug.

[`amz_s3_endpoint`](https://github.com/JuliaCloud/AWS.jl/blob/4ca32795fb3e7d4a98f9b4299963fff9c65192a4/src/S3.jl#L909) seems to be ignoring endpoint URL set in `AWSEnv`.

**Comment by akubisch** [comment]
Alright - I've fixed it for the moment by hardcoding my endpoint in `amz_s3_endpoint` in S3.jl, but it would be great if that could be resolved.

**Comment by tanmaykm** [comment]
https://github.com/JuliaCloud/AWS.jl/pull/112 should fix this.

@akubisch could you check if it does?

**Comment by akubisch** [comment]
Indeed it does, thank you very much for the swift fix!

**Comment by tanmaykm** [comment]
Thanks for filling the issue!

**Comment by visr** [comment]
Should this issue now be reopened? It seems that the endpoint is fixed to amazonaws.com: https://github.com/JuliaCloud/AWS.jl/blob/5fcce52ed2f46fcd9e600027662eb8693bbef043/src/AWS.jl#L445. Bumping into this in https://github.com/meggart/Zarr.jl/issues/11.

**Comment by mattBrzezinski** [comment]
Yes I'll open this issue up. I remember this occurring in `AWSCore.jl`, but since then I had forgotten this was a thing as I don't use it.

**Comment by mattBrzezinski** [comment]
FIx for this is here, https://github.com/JuliaCloud/AWS.jl/pull/211

**Comment by altre** [comment]
Any chance of this making it into a release any time soon?

**Comment by mattBrzezinski** [comment]
> Any chance of this making it into a release any time soon?

This should have been resolved in `AWS@1.25.0`, I have not had the time to check it. But that `AbstractAWSConfig` should enable you to resolve this.
